warehouse/templates/includes/input-recaptcha.html

```diff
@@ -0,0 +1,29 @@
+{% macro recaptcha_html() -%}
+  {% if request.find_service(name="recaptcha").enabled %}
+    <div id="recaptcha-container" class="g-recaptcha"></div>
+    {% if form.g_recaptcha_response.errors %}
+    <ul class="errors">
+      {% for error in form.g_recaptcha_response.errors %}
+      <li>{{ error }}</li>
+      {% endfor %}
+    </ul>
+    {% endif %}
+  {% endif %}
+{%- endmacro %}
+
+{% macro recaptcha_src() -%}
+  {% if request.find_service(name="recaptcha").enabled %}
+    <script type="text/javascript">
+      var config = new Object();
+      config.sitekey = "{{ request.registry.settings["recaptcha"]["site_key"] }}";
```

Comment:
I think that having this inline script here is generally not the way we want to go about this. We should be able to do something like:

```
<script
    id="recaptcha-js"
    src="https://www.google.com/recaptcha/api.js?onload=renderCaptcha&render=explicit"
    data-site-key="{{ request.registry.settings["recaptcha"]["site_key"] }}"
    async
    defer>
</script>
```

Then in our static JS files we can do something like:

```javascript
renderCaptcha = function() {
    var config = new Object();
    config.sitekey = $("script#recaptcha-js").data("site-key");
    grecaptcha.render($("#recaptcha-container"), config);
};
```

This should make it so that we remove all of the inline scripts (which protects against XSS, since we won't need unsafe-inline in the CSP policy) and doesn't rely on using HTML escaping inside of a javascript block.